### PR TITLE
Http header deprecated

### PR DESCRIPTION
**Pull request by psyren89**
As of API22, calls to Android's internal version of `org.apache.http.Header` are deprecated (see links).
Using the latest version of Apache's `HttpClient` fixes this issue.

While I was at it, I also fixed a deprecated call to `URLDecoder.decode()`.

https://developer.android.com/reference/org/apache/http/Header.html
http://android-developers.blogspot.com.au/2011/09/androids-http-clients.html

### CODE REVIEW

**Comment by wei-lee** [comment]
reviewed. Looks good but the build failed due to this error: java.lang.IncompatibleClassChangeError

**Comment by psyren89** [comment]
Most of what I've read implies that recompiling should fix the issue. Perhaps `clean` needs to be added before every `install` command in the Maven build script?

**Comment by psyren89** [comment]
Only one build task actually failed with that error - the other two timed out waiting for the emulator.

**Comment by psyren89** [comment]
Bah, still can't get it to build.
The funny thing is, it works fine on my machine: 
screen shot 2015-08-14 at 10 24 32 am

**Comment by danielpassos** [comment]
@psyren89 I'll give it a try tomorrow morning

**Comment by psyren89** [comment]
Squashed the commits for easier reviewing.

**Comment by danielpassos** [comment]
I'm getting the same of @travis-ci

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Feedhenry Android SDK Parent ...................... SUCCESS [0.200s]
[INFO] Feedhenry Android SDK Library ..................... SUCCESS [10.582s]
[INFO] FeedHenry Android SDK Test ........................ FAILURE [1:08.033s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1:20.383s
[INFO] Finished at: Fri Aug 14 09:24:09 BRT 2015
[INFO] Final Memory: 41M/180M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.simpligility.maven.plugins:android-maven-plugin:4.1.1:internal-integration-test (default-internal-integration-test) on project fh-android-sdk-test: 192.168.60.101_5555_Genymotion_GoogleNexus5-5.0.0-API21-1080x1920 :   Test run failed to complete: Instrumentation run failed due to 'java.lang.IncompatibleClassChangeError' -> [Help 1]
```

I'm also have some new warnings

```
[INFO] Convert classes to Dex : /Users/passos/Development/Projects/FeedHenry/fh-android-sdk/fh-android-sdk-test/target/classes.dex
[INFO] warning: Ignoring InnerClasses attribute for an anonymous inner class
[INFO] (org.apache.commons.logging.impl.WeakHashtable$1) that doesn't come with an
[INFO] associated EnclosingMethod attribute. This class was probably produced by a
[INFO] compiler that did not target the modern .class file format. The recommended
[INFO] solution is to recompile the class from source, using an up-to-date compiler
[INFO] and without specifying any "-target" type options. The consequence of ignoring
[INFO] this warning is that reflective operations on this class will incorrectly
[INFO] indicate that it is *not* an inner class.
[INFO] warning: Ignoring InnerClasses attribute for an anonymous inner class
[INFO] (org.apache.commons.logging.impl.LogFactoryImpl$1) that doesn't come with an
[INFO] associated EnclosingMethod attribute. This class was probably produced by a
[INFO] compiler that did not target the modern .class file format. The recommended
[INFO] solution is to recompile the class from source, using an up-to-date compiler
[INFO] and without specifying any "-target" type options. The consequence of ignoring
[INFO] this warning is that reflective operations on this class will incorrectly
[INFO] indicate that it is *not* an inner class.
[INFO] warning: Ignoring InnerClasses attribute for an anonymous inner class
[INFO] (org.apache.commons.logging.impl.LogFactoryImpl$3) that doesn't come with an
[INFO] associated EnclosingMethod attribute. This class was probably produced by a
[INFO] compiler that did not target the modern .class file format. The recommended
[INFO] solution is to recompile the class from source, using an up-to-date compiler
[INFO] and without specifying any "-target" type options. The consequence of ignoring
[INFO] this warning is that reflective operations on this class will incorrectly
[INFO] indicate that it is *not* an inner class.
[INFO] warning: Ignoring InnerClasses attribute for an anonymous inner class
[INFO] (org.apache.commons.logging.impl.SimpleLog$1) that doesn't come with an
[INFO] associated EnclosingMethod attribute. This class was probably produced by a
[INFO] compiler that did not target the modern .class file format. The recommended
[INFO] solution is to recompile the class from source, using an up-to-date compiler
[INFO] and without specifying any "-target" type options. The consequence of ignoring
[INFO] this warning is that reflective operations on this class will incorrectly
[INFO] indicate that it is *not* an inner class.
[INFO] warning: Ignoring InnerClasses attribute for an anonymous inner class
[INFO] (org.apache.commons.logging.LogFactory$4) that doesn't come with an
[INFO] associated EnclosingMethod attribute. This class was probably produced by a
[INFO] compiler that did not target the modern .class file format. The recommended
[INFO] solution is to recompile the class from source, using an up-to-date compiler
[INFO] and without specifying any "-target" type options. The consequence of ignoring
[INFO] this warning is that reflective operations on this class will incorrectly
[INFO] indicate that it is *not* an inner class.
[INFO] warning: Ignoring InnerClasses attribute for an anonymous inner class
[INFO] (org.apache.commons.logging.LogFactory$3) that doesn't come with an
[INFO] associated EnclosingMethod attribute. This class was probably produced by a
[INFO] compiler that did not target the modern .class file format. The recommended
[INFO] solution is to recompile the class from source, using an up-to-date compiler
[INFO] and without specifying any "-target" type options. The consequence of ignoring
[INFO] this warning is that reflective operations on this class will incorrectly
[INFO] indicate that it is *not* an inner class.
[INFO] warning: Ignoring InnerClasses attribute for an anonymous inner class
[INFO] (org.apache.commons.logging.LogFactory$6) that doesn't come with an
[INFO] associated EnclosingMethod attribute. This class was probably produced by a
[INFO] compiler that did not target the modern .class file format. The recommended
[INFO] solution is to recompile the class from source, using an up-to-date compiler
[INFO] and without specifying any "-target" type options. The consequence of ignoring
[INFO] this warning is that reflective operations on this class will incorrectly
[INFO] indicate that it is *not* an inner class.
[INFO] warning: Ignoring InnerClasses attribute for an anonymous inner class
[INFO] (org.apache.commons.logging.LogFactory$5) that doesn't come with an
[INFO] associated EnclosingMethod attribute. This class was probably produced by a
[INFO] compiler that did not target the modern .class file format. The recommended
[INFO] solution is to recompile the class from source, using an up-to-date compiler
[INFO] and without specifying any "-target" type options. The consequence of ignoring
[INFO] this warning is that reflective operations on this class will incorrectly
[INFO] indicate that it is *not* an inner class.
[INFO] warning: Ignoring InnerClasses attribute for an anonymous inner class
[INFO] (org.apache.commons.logging.impl.LogFactoryImpl$2) that doesn't come with an
[INFO] associated EnclosingMethod attribute. This class was probably produced by a
[INFO] compiler that did not target the modern .class file format. The recommended
[INFO] solution is to recompile the class from source, using an up-to-date compiler
[INFO] and without specifying any "-target" type options. The consequence of ignoring
[INFO] this warning is that reflective operations on this class will incorrectly
[INFO] indicate that it is *not* an inner class.
[INFO] warning: Ignoring InnerClasses attribute for an anonymous inner class
[INFO] (org.apache.commons.logging.LogFactory$1) that doesn't come with an
[INFO] associated EnclosingMethod attribute. This class was probably produced by a
[INFO] compiler that did not target the modern .class file format. The recommended
[INFO] solution is to recompile the class from source, using an up-to-date compiler
[INFO] and without specifying any "-target" type options. The consequence of ignoring
[INFO] this warning is that reflective operations on this class will incorrectly
[INFO] indicate that it is *not* an inner class.
[INFO] warning: Ignoring InnerClasses attribute for an anonymous inner class
[INFO] (org.apache.commons.logging.LogFactory$2) that doesn't come with an
[INFO] associated EnclosingMethod attribute. This class was probably produced by a
[INFO] compiler that did not target the modern .class file format. The recommended
[INFO] solution is to recompile the class from source, using an up-to-date compiler
[INFO] and without specifying any "-target" type options. The consequence of ignoring
[INFO] this warning is that reflective operations on this class will incorrectly
[INFO] indicate that it is *not* an inner class.
```

**Comment by psyren89** [comment]
This latest commit should fix the build issues.

**Comment by psyren89** [comment]
Something is very wrong with the CI setup. I've followed the script set out in `.travis.yml` and fixed the errors that CI used to return, yet Travis still doesn't want to play nice.

**Comment by danielpassos** [comment]
I'll test locally and take a look at @travis-ci

**Comment by danielpassos** [comment]
@psyren89 @travis-ci problem fixed expected by Java6. Let me send a PR to drop it, before we merge this

**Comment by danielpassos** [comment]
@psyren89 Just send a [PR](https://github.com/thefrontiergroup/fh-android-sdk/pull/9) to your branch

**Comment by psyren89** [comment]
It turns out the "wait for emulator" script waits for the boot animation to finish before completing...

File: .travis.yml [hunk 1]
```diff
@@ -48,9 +47,9 @@ before_script:
 
 script:
   - set -ev
-  - mvn install -Drelease
+  - mvn clean install -Drelease
```
Comment (danielpassos) [inline .travis.yml]: Any reason to add clean before install? Keep in mind the travis always clean the machine before run the tests

**Comment by danielpassos** [comment]
Please don't kill the provided AeroGear Android Push jar it's provided to make some IDE's happy

**Comment by psyren89** [comment]
Do you mean Eclipse? I'll test whether it still works and revert it if it doesn't.

**Comment by danielpassos** [comment]
@psyren89 NetBeans for example. @secondsun can give you more details. Anyway I can see any benefit in remove it.

File: .travis.yml (hunk 1 above)
Comment (psyren89) [inline .travis.yml]: It can be argued that it's good practise but if you feel it's wholly unnecessary it can be removed.

Comment (danielpassos) [inline .travis.yml]: Chilax, just checking if you had any experience with it on @travis-ci. No problem for me.

Comment (psyren89) [inline .travis.yml]: :+1:

**Comment by psyren89** [comment]
Aha, right. I'll put it back in with a note.

**Comment by danielpassos** [comment]
:+1:

**Comment by psyren89** [comment]
Good to merge?

**Comment by wei-lee** [comment]
merged in 5d9713bc82fe4b435ebcd1adbaa7abc4d3aaed17

**Pull request closed without merging**